ccl/backupccl: TestClusterRestoreFailCleanup failed

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/474277c7-a2a4-482f-93e1-f5730939539c?testReportRun=9&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [097880977a1f5bfb8377afd2a407cdf7ca4fe377](https://github.com/cockroachdb/cockroach/commits/097880977a1f5bfb8377afd2a407cdf7ca4fe377):


```
=== RUN   TestClusterRestoreFailCleanup
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestClusterRestoreFailCleanup1329344848
    test_log_scope.go:81: use -show-logs to present logs inline
    testutils.go:289: no Invalid Descriptors
    full_cluster_backup_restore_test.go:901: -- test log scope end --
test logs left over in: outputs.zip/logTestClusterRestoreFailCleanup1329344848
--- FAIL: TestClusterRestoreFailCleanup (65.27s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (41.15s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/tenant_settings
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:87: error executing query="SET CLUSTER SETTING jobs.registry.retry.initial_delay = '1ms'" args=[]: pq: setting updated but timed out waiting to read new value
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/tenant_settings (12.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34808

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/389e6840-3599-499a-b0e2-9cd00230a383?testReportRun=3&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [e2e88aa288c81282b20ca0640da983f97b234324](https://github.com/cockroachdb/cockroach/commits/e2e88aa288c81282b20ca0640da983f97b234324):


```
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/scheduled_jobs (179.67s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (1059.97s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:363
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:366
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:363
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members (149.47s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/8d5b8c67-35b6-4168-965f-87485669da8d?testReportRun=2&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [ff291e3ee70eedac50b985860502d2f0b0867003](https://github.com/cockroachdb/cockroach/commits/ff291e3ee70eedac50b985860502d2f0b0867003):


```
=== RUN   TestClusterRestoreFailCleanup
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestClusterRestoreFailCleanup1011735475
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testutils.go:289: no Invalid Descriptors
    full_cluster_backup_restore_test.go:901: -- test log scope end --
test logs left over in: outputs.zip/logTestClusterRestoreFailCleanup1011735475
--- FAIL: TestClusterRestoreFailCleanup (1038.96s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (743.39s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/users
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/users (135.81s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/3784f56c-a4ac-49a5-931b-f8831df67029?testReportRun=1&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [d2d974202781321ddb4fb1115c68fe7566ab9d06](https://github.com/cockroachdb/cockroach/commits/d2d974202781321ddb4fb1115c68fe7566ab9d06):


```
=== RUN   TestClusterRestoreFailCleanup
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestClusterRestoreFailCleanup3105122427
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testutils.go:289: no Invalid Descriptors
    full_cluster_backup_restore_test.go:901: -- test log scope end --
test logs left over in: outputs.zip/logTestClusterRestoreFailCleanup3105122427
--- FAIL: TestClusterRestoreFailCleanup (1608.29s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_id_seq
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_id_seq (150.58s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (1392.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/9e314f26-e389-4c63-9066-d75bf5b1bb33?testReportRun=1&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (135.65s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/scheduled_jobs
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/scheduled_jobs (127.62s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/05b0e99b-d107-4e85-a3f1-e9a9d88b8559?testReportRun=3&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/privileges (227.39s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (249.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/fd2b1628-6e2d-4c90-a5c2-a9a4610deb39?testReportRun=1&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/users (318.41s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (2474.00s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_id_seq
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_id_seq (479.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/3c84960c-5927-477f-881f-5c6154b5b97d?testReportRun=3&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/scheduled_jobs (174.55s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/users
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/users (168.44s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/3248e418-de91-40d2-871c-3359f9d01e35?testReportRun=1&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [4f31fbae529e459c00ef2490e193c0c246e8d0f9](https://github.com/cockroachdb/cockroach/commits/4f31fbae529e459c00ef2490e193c0c246e8d0f9):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members (196.10s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (168.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/9a7a0fad-4521-4f50-8729-bf6d8f8789cb?testReportRun=3&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [57bd0a92676b1c58ee7e29979ce3dceb5837abdb](https://github.com/cockroachdb/cockroach/commits/57bd0a92676b1c58ee7e29979ce3dceb5837abdb):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/database_role_settings (169.39s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (200.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/8c1cff8d-bd76-4a94-ba94-492fc11003b7?testReportRun=3&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (246.74s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/settings
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/settings (257.82s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/f4a84354-673d-45b5-87e7-b12209e8eb4b?testReportRun=1&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members (226.19s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (155.98s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (912.45s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/e3897ad1-3bd1-4605-888d-d6535e862765?testReportRun=3&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (112.05s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/scheduled_jobs
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/scheduled_jobs (105.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/8d7319c1-b038-4e2a-87c6-2e99de81e84a?testReportRun=3&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members (144.81s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (175.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/6f2cff4d-c2d3-4903-8f72-e74af552ca31?testReportRun=2&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/database_role_settings (157.77s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_members (225.31s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestClusterRestoreFailCleanup [failed](https://tanzanite.cluster.engflow.com/invocations/default/74e807d2-2fb6-494d-9dce-f748f9ec8fd4?testReportRun=2&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [435a330ddf15f70234747ef55d3d3721a09ba252](https://github.com/cockroachdb/cockroach/commits/435a330ddf15f70234747ef55d3d3721a09ba252):


```
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/role_options (184.43s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore
    --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore (1212.12s)
=== RUN   TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/external_connections
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:120: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            				github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	Error:      	Received unexpected error:
        	            	query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:371
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:133
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/backupccl.TestClusterRestoreFailCleanup.func6.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/backupccl/full_cluster_backup_restore_test.go:811
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1595
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1650
        	            	Wraps: (2) query 'SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'': expected:
        	            	  | 1
        	            	  |
        	            	  | got:
        	            	  | 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    testutils.go:289: no Invalid Descriptors
        --- FAIL: TestClusterRestoreFailCleanup/retry-during-custom-system-table-restore/external_connections (184.87s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterRestoreFailCleanup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
+++ b/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
@@ -776,8 +776,9 @@ func TestClusterRestoreFailCleanup(t *testing.T) {
 		}
 		for customRestoreSystemTable, cfg := range customRestoreSystemTables {
 			t.Run(customRestoreSystemTable, func(t *testing.T) {
+				short := time.Millisecond * 5
 				args := base.TestClusterArgs{ServerArgs: base.TestServerArgs{
-					Knobs: base.TestingKnobs{JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals()},
+					Knobs: base.TestingKnobs{JobsTestingKnobs: jobs.NewTestingKnobsWithIntervals(short, short, time.Microsecond, time.Millisecond)},
 				}}
 				tcRestore, sqlDBRestore, cleanupEmptyCluster := backupRestoreTestSetupEmpty(t, singleNode, tempDir, InitManualReplication, args)
 				defer cleanupEmptyCluster()
@@ -805,12 +806,10 @@ func TestClusterRestoreFailCleanup(t *testing.T) {
 				}
 				// The initial restore will return an error, and restart.
 				sqlDBRestore.ExpectErr(t, `running execution from '.*' to '.*' on \d+ failed: injected error`, `RESTORE FROM LATEST IN $1`, localFoo)
-				// Reduce retry delays.
-				sqlDBRestore.Exec(t, "SET CLUSTER SETTING jobs.registry.retry.initial_delay = '1ms'")
+				tcRestore.Servers[0].JobRegistry().(*jobs.Registry).TestingNudgeAdoptionQueue()
 				// Expect the restore to succeed.
-				sqlDBRestore.CheckQueryResultsRetry(t,
-					`SELECT count(*) FROM [SHOW JOBS] WHERE job_type = 'RESTORE' AND status = 'succeeded'`,
-					[][]string{{"1"}})
+				id := sqlDBRestore.QueryStr(t, `SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'RESTORE' ORDER BY created DESC LIMIT 1`)[0][0]
+				sqlDBRestore.CheckQueryResultsRetry(t, fmt.Sprintf(`SELECT status FROM [SHOW JOB WHEN COMPLETE %s]`, id), [][]string{{"succeeded"}})
 			})
 		}
 	})

